lib/DxilPIXPasses/DxilPIXAddTidToAmplificationShaderPayload.cpp

```diff
@@ -35,19 +35,24 @@ class DxilPIXAddTidToAmplificationShaderPayload : public ModulePass {
   const char *getPassName() const override { return "DXIL Add flat thread id to payload from AS to MS"; }
   bool runOnModule(Module &M) override;
   void applyOptions(PassOptions O) override;
-
-private:
-
-    void EmitInstructionsToCopyStructContents(llvm::IRBuilder<> &B,
-        llvm::Value *NewStructAlloca,
-        llvm::Value*OldStructAlloca);
 };
 
 void DxilPIXAddTidToAmplificationShaderPayload::applyOptions(PassOptions O) {
   GetPassOptionUInt32(O, "dispatchArgY", &m_DispatchArgumentY, 1);
   GetPassOptionUInt32(O, "dispatchArgZ", &m_DispatchArgumentZ, 1);
 }
 
+void AddValueToExpandedPayload(OP* HlslOP, llvm::IRBuilder<> &B, ExpandedStruct & expanded, AllocaInst* NewStructAlloca, unsigned int expandedValueIndex, Value* value) {
+  Constant *Zero32Arg = HlslOP->GetU32Const(0);
+  SmallVector<Value *, 2> IndexToAppendedValue;
+  IndexToAppendedValue.push_back(Zero32Arg);
+  IndexToAppendedValue.push_back(HlslOP->GetU32Const(expandedValueIndex));
```

Comment:
Pre-existing, but since you have the values you want to populate this SmallVector with,  you could have the same effect with ` SmallVector<Value *, 2> IndexToAppendedValue(Zero32Arg, HlslOP->GetU32Const(expandedValueIndex));` if you wanted. Entirely unnecessary though.

Reply:
I think I'll leave it due to the testing rigamarole that would ensue. Thanks for the education.